client/signup/steps/design-type/index.jsx

```diff
@@ -1,73 +1,99 @@
+			{ type: 'blog', label: translate( 'A list of my latest posts' ), image: <BlogImage /> },
+			{ type: 'page', label: translate( 'A welcome page for my site' ), image: <PageImage /> },
+			{ type: 'grid', label: translate( 'A grid of my latest posts' ), image: <GridImage /> },
 		];
-	},
-
-	renderChoice( choice ) {
-		return (
-			<Card className="design-type__choice" key={ choice.type }>
-				<a className="design-type__choice__link" href="#" onClick={ ( event ) => this.handleChoiceClick( event, choice.type ) }>
-					{ choice.image }
-					<h2>{ choice.label }</h2>
-				</a>
-			</Card>
-		);
-	},
+	}
 
 	renderChoices() {
 		return (
 			<div className="design-type__list">
-				{ this.getChoices().map( this.renderChoice ) }
+				{ this.getChoices().map( ( choice ) => (
+						<Card className="design-type__choice" key={ choice.type }>
+							<a
+								className="design-type__choice-link"
+								ref="#" onClick={ ( event ) => this.handleChoiceClick( event, choice.type ) }
+							>
+								{ choice.image }
+								<h2>{ choice.label }</h2>
+							</a>
+						</Card>
+					)
+				) }
 				<div className="design-type__choice is-spacergif" />
 			</div>
 		);
-	},
+	}
 
 	render() {
+		const { translate } = this.props;
 		return (
-			<div className="design-type__section-wrapper">
+			<div className="design-type">
 				<StepWrapper
 					flowName={ this.props.flowName }
 					stepName={ this.props.stepName }
 					positionInFlow={ this.props.positionInFlow }
-					fallbackHeaderText={ this.translate( 'What would you like your homepage to look like?' ) }
-					fallbackSubHeaderText={ this.translate( 'This will help us figure out what kinds of designs to show you.' ) }
-					subHeaderText={ this.translate( 'First up, what would you like your homepage to look like?' ) }
+					fallbackHeaderText={ translate( 'What would you like your homepage to look like?' ) }
+					fallbackSubHeaderText={ translate( 'This will help us figure out what kinds of designs to show you.' ) }
+					subHeaderText={ translate( 'First up, what would you like your homepage to look like?' ) }
 					signupProgress={ this.props.signupProgress }
 					stepContent={ this.renderChoices() } />
 			</div>
 		);
-	},
+	}
 
 	handleChoiceClick( event, type ) {
 		event.preventDefault();
 		event.stopPropagation();
 		this.handleNextStep( type );
-	},
+	}
 
 	handleNextStep( designType ) {
-		analytics.tracks.recordEvent( 'calypso_triforce_select_design', { category: designType } );
+		this.props.recordChosenDesignType( designType );
 
 		SignupActions.submitSignupStep( { stepName: this.props.stepName }, [], { designType } );
 		this.props.goToNextStep();
 	}
-} );
+}
+
+export default connect(
+	null,
+	( dispatch ) => ( {
+		recordChosenDesignType: ( designType ) => {
+			dispatch( recordTracksEvent( 'calypso_triforce_select_design', { category: designType } ) );
```

Comment:
Consider splitting this into:

- the pure action creator:

```js
const recordChosenDesignType = ( designType ) =>
  recordTracksEvent( … );
```

- the binding of the dispatcher:

```js
connect( null, { recordChosenDesignType } )( localize( DesignTypeStep ) );
```

It reads better and is potentially more performant.